gcloud/storage/blob.py

```diff
@@ -64,23 +77,18 @@ class Blob(_PropertyMixin):
     # ACL rules are lazily retrieved.
     _acl = None
 
-    def __init__(self, bucket=None, name=None, properties=None):
-        """Blob constructor.
-
-        :type bucket: :class:`gcloud.storage.bucket.Bucket`
-        :param bucket: The bucket to which this blob belongs.
-
-        :type name: string
-        :param name: The name of the blob.  This corresponds to the
-                     unique path of the object in the bucket.
-
-        :type properties: dict
-        :param properties: All the other data provided by Cloud Storage.
-        """
+    def __init__(self, name, bucket=None, properties=None):
         if name is None and properties is not None:
             name = properties.get('name')
+
         super(Blob, self).__init__(name=name, properties=properties)
+
+        if bucket is None:
+            bucket = _implicit_environ.BUCKET
+
         self.bucket = bucket
+        if bucket is None:
```

Comment:
I am hesitant to do this because

1. At some point, something like `bucket.connection` or `bucket.get_blob`, etc. will fail
1. Importing `Bucket` here and `Blob` in `bucket.py` makes it harder to manage interdependence of modules.